### PR TITLE
dist/tools: add sort-variables script

### PR DESCRIPTION
**Pull request by vincent-d**
This is the script I used for #6740.

It's not really tested and comes without any warranty as it might have some side effects.

I thought it might still be interesting to keep it, though.

### CODE REVIEW

**Comment by miri64** [comment]
Would be great if this tool could also check *if* the variables are sorted.

**Comment by miri64** [comment]
(for CI)

**Comment by miri64** [comment]
@vincent-d ping?

**Comment by vincent-d** [comment]
Sorry, this was not in my priority list ;)

I don't know if I'll have time to work on this soon, but feel free to improve if you want.

**Comment by miri64** [comment]
Okay, but I guess it is a start :-)

**Comment by vincent-d** [comment]
It can be easy to remove comments before sorting, but I don't see an easy way to sort and keep comments...

**Comment by miri64** [comment]
Any plans to work on this further? Maybe we can come up with a solution to keep the comments or agree on a scheme how to comment on a list otherwise.

File: dist/tools/sort-variables/sort-variables.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+
+SEARCH_PATH=$1
+VAR=$2
+TMPFILE=$(mktemp)
+
+if [ $# -lt 2 ]; then
+    echo "Usage: sort-variables.sh <path> <variable>"
+    echo "Sort the <variable> alphabetically, and cut lines to 80 characters"
+    echo "It will search is <path> and its subdirectories"
```
[the remark this answers is not in the document]
Comment (smlng) [reply dist/tools/sort-variables/sort-variables.sh]: or should be `in`?

**Comment by miri64** [comment]
Ping @vincent-d?

**Comment by tcschmidt** [comment]
@vincent-d how would you like to proceed here? This seems to require some strategic effort ;)

**Comment by miri64** [comment]
Ping @vincent-d. Can this be closed in favor of #10111?

**Comment by OlegHahm** [comment]
ping @vincent-d again

**Comment by vincent-d** [comment]
Sorry for being unresponsive here. 

I still don't have time to invest in this. As I said, I guess we could remove comments when sorting. If everyone can live with this, then we could merge.

**Pull request closed without merging**